File: TAREAS 5 SEMESTRE/Portafolio SO/programas/programa6/Nodo.java
package listaSimple;

public class Nodo {
	public int entidad;
	public boolean consumible;
	public Nodo sig;
	public int index;
	
	Nodo(){
		this.entidad = -1;
		this.consumible = false;
		this.sig = null;
	}
	
	Nodo(int index){
		this.entidad = -1;
		this.consumible = false;
		this.sig = null;
		this.index = index;
	}
	
	Nodo(Nodo sig, int index){
		this.entidad = -1;
		this.consumible = false;
		this.sig = sig;
		this.index = index;
	}
}

File: TAREAS 4 SEMESTRE/Plantillas/InsertionSortAngel/src/Nodo.cpp
#include "Nodo.h"

Nodo::Nodo()
{
    sig=nullptr;
    ant=nullptr;
    dato=0;
}
Nodo::Nodo(Nodo* sig,Nodo* ant)
{
    this->sig=sig;
    this->ant=ant;
    dato=0;
}
Nodo::Nodo(int dato)
{
    sig=nullptr;
    ant=nullptr;
    this->dato=dato;
}
Nodo::Nodo(int dato,Nodo* sig,Nodo* ant)
{
    this->dato=dato;
    if(!sig)
        this->sig=nullptr;
    else
        this->sig=sig;
    if(!ant)
        this->ant=nullptr;
    else
        this->ant=ant;
}
